File: HorseRacing.Contracts/Models/Game/Responses/GetLobbyUsersWithBets/GetLobbyUsersWithBetsResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Game.Responses.GetLobbyUsersWithBets
{
    public class GetLobbyUsersWithBetsResponse : BaseResponse<GetLobbyUsersWithBetsResponseDto>
    {
        public GetLobbyUsersWithBetsResponse() : base() { }
        public GetLobbyUsersWithBetsResponse(GetLobbyUsersWithBetsResponseDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/get-lobby-users-with-bets/get-lobby-users-with-bets-response.ts
import type { BaseResponse } from "~/interfaces/api/contracts/base/responses/base-response";
import type { GetLobbyUsersWithBetsResponseDto } from "./get-lobby-users-with-bets-response-dto";

export class GetLobbyUsersWithBetsResponse implements BaseResponse<GetLobbyUsersWithBetsResponseDto> {
    Data: GetLobbyUsersWithBetsResponseDto;

    constructor(data?: GetLobbyUsersWithBetsResponseDto) {
        this.Data = data || { GameId: '', GameName: '', Players: []};
    }
}
